bokeh/io/export.py

```diff
@@ -119,7 +119,7 @@ def export_png(obj: LayoutDOM | Document, *, filename: PathLike | None = None, w
     filename = os.fspath(filename) # XXX: Image.save() doesn't fully support PathLike
     image.save(filename)
 
-    return abspath(filename)
+    return abspath(expanduser(filename))
```

Comment:
`Image.save()` does not work with `~`, or more specifically `open` does not work with `~`. 

I can remove the `expanduser` if this is the desired behavior.

``` python
import os

from bokeh.plotting import figure
from bokeh.io import export_png


p = figure(x_range=(0, 100), y_range=(0, 100), toolbar_location=None)
export_png(p, filename=os.path.expanduser("~/plot.png"))  # Works
export_png(p, filename="~/plot.png")  # Raises FileNotFoundError
```

``` python-traceback
Traceback (most recent call last):
  File "/home/shh/folder with space/save.py", line 7, in <module>
    export_png(p, filename="~/plot.png")
  File "/home/shh/Downloads/bokeh/bokeh/io/export.py", line 120, in export_png
    image.save(filename)
  File "/home/shh/miniconda3/envs/tmp_bokeh/lib/python3.8/site-packages/Pillow-9.1.1-py3.8-linux-x86_64.egg/PIL/Image.py", line 2297, in save
    fp = builtins.open(filename, "w+b")
FileNotFoundError: [Errno 2] No such file or directory: '~/plot.png'
```